src/commands/registers.ts

```diff
@@ -222,10 +234,386 @@ export class SomeRegisterCommand extends EmacsCommand {
+      return;
+    }
+
+    if (!data.positions || data.positions.length === 0) {
+      MessageManager.showMessage("No position stored in this register");
+      return;
+    }
+
+    // Get positions from register and ensure they are valid
+    const storedPositions = data.positions;
+    if (!storedPositions || storedPositions.length === 0) {
+      MessageManager.showMessage("No positions stored in register");
+      return;
+    }
+
+    // First, save current positions as marks before moving
+    const currentMarks = currentPositions.map((pos) => new vscode.Position(pos.line, pos.character));
+
+    // Ensure we have at least one mark
+    if (currentMarks.length === 0) {
+      return;
+    }
+
+    // Create selections using the stored positions, ensuring type safety
+    const validStoredPositions = storedPositions.filter((pos): pos is vscode.Position => {
+      return pos instanceof vscode.Position && typeof pos.line === "number" && typeof pos.character === "number";
+    });
+    if (validStoredPositions.length === 0) {
+      MessageManager.showMessage("Invalid positions in register");
+      return;
+    }
+
+    // Use the first stored position for all selections to maintain consistent behavior
+    // We can use non-null assertion here because we've verified the array has elements
+    const targetPosition: vscode.Position = validStoredPositions[0]!;
+
+    // Save current positions as marks before moving
+    const marks = currentPositions.map((pos) => new vscode.Position(pos.line, pos.character));
+
+    // Set the marks for exchange point and mark functionality
+    // Important: These marks should be the positions we're jumping FROM
+    this.emacsController.pushMark(marks);
+    this.emacsController.setPrevExchangedMarks(marks);
+
+    // Create selections at target position with proper mark handling
+    // When jumping to a position in Emacs, we want to:
+    // 1. Set the mark at the current position (already done with pushMark above)
+    // 2. Move to the target position
+    // 3. Create a selection that spans from mark to target position
+    const currentPos = currentPositions[0];
+    if (!currentPos) {
+      return;
+    }
+
+    // In Emacs, when jumping to a register:
+    // - The current position becomes the mark (anchor point)
+    // - The target position becomes the active point (where the cursor is)
+    // First, set the mark at the current position
+    this.emacsController.pushMark([currentPos]);
+    this.emacsController.enterMarkMode();
+
+    // Create a selection from the current position to the target position
+    // The active position (cursor) should be at the target
+    const newSelection = new vscode.Selection(currentPos, targetPosition);
+    textEditor.selections = [newSelection];
+
+    // Reveal cursor and notify user
+    revealPrimaryActive(textEditor);
+    MessageManager.showMessage("Jumped to saved position");
+  }
+}
+
+// copy-to-register, C-x r s <r>
```

Comment:
What are these commands, `CopyToRegister`, `CopyRectangleToRegister`, `PointToRegister`, `JumpToRegister`, and `InsertRegister`?
Carefully consider what "pre-` commands and `SomeRegisterCommand` do. All commands are implemented in `SomeRegisterCommand.run-`.
Don't add such a trash.